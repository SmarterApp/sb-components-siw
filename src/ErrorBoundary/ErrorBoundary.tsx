import * as React from "react";

export interface ErrorBoundaryProps {
  fallbackUI: JSX.Element;
}

export interface ErrorBoundaryState {
  hasError: boolean;
  error?: Error;
  // tslint:disable-next-line: no-any
  errorInfo?: any;
}

export class ErrorBoundary extends React.Component<
  ErrorBoundaryProps,
  ErrorBoundaryState
> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  // tslint:disable-next-line: no-any
  componentDidCatch(error: Error, info: any) {
    // You can log the error to an error reporting service here
    // Display fallback UI
    this.setState({ error, hasError: true, errorInfo: info });
  }

  render(): JSX.Element | React.ReactNode {
    return this.state.hasError ? this.props.fallbackUI : this.props.children;
  }
}
