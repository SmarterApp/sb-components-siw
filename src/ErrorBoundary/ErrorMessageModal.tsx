import * as React from "react";
import * as ReactModal from "react-modal";
export interface ErrorMessageModalProps {
  showModal: boolean;
  StatusMessage: string;
  onChangeErrorModelState: () => void;
}
export class ErrorMessageModal extends React.Component<ErrorMessageModalProps> {
  constructor(props: ErrorMessageModalProps) {
    super(props);
  }
  handleHideModal = () => {
    this.props.onChangeErrorModelState();
  };
  render() {
    const modelState = this.props.showModal;
    return (
      <div className="search-result-container">
        <ReactModal
          isOpen={modelState}
          contentLabel="About This Item Modal"
          onRequestClose={this.handleHideModal}
          overlayClassName="react-modal-overlay"
          className="react-modal-content about-item-modal"
        >
          <div
            className="modal-wrapper"
            aria-labelledby="About Item Modal"
            aria-hidden="true"
          >
            <div className="modal-header">
              <h4 className="modal-title">
                <h4> </h4>
              </h4>
              <button
                className="close"
                onClick={this.handleHideModal}
                aria-label="Close modal"
              >
                <span className="fa fa-times" aria-hidden="true" />
              </button>
            </div>
            <div className="modal-body">
              <form id="accessibility-form">
                <div className="accessibility-groups">
                  <div className="accessibility-resource-type section section-light">
                    <h5 style={{ fontWeight: "bold" }}>
                      <i
                        className="error-message fa fa-exclamation-triangle"
                        aria-hidden="true"
                      />
                      {"    "}

                      {this.props.StatusMessage}
                    </h5>
                  </div>
                </div>
              </form>
            </div>
            <div className="modal-footer">
              <button
                className="btn btn-primary"
                aria-label="Close modal"
                onClick={this.handleHideModal}
              >
                Ok
              </button>
            </div>
          </div>
        </ReactModal>
      </div>
    );
  }
}
