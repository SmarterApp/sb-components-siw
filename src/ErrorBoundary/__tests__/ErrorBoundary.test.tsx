import * as React from "react";
import * as ReactDOM from "react-dom";
import "jsdom-global/register";
import { shallow, mount, render } from "enzyme";
import { ErrorBoundary } from "../ErrorBoundary";

// tslint:disable: variable-name
const ErrorMssg: React.SFC = () => {
  return <div>An error occured.</div>;
};

const ErrorThrower: React.SFC = () => {
  const handleClick = () => {
    throw new Error("Test Error");
  };

  return (
    <div>
      <button className="test-button" onClick={() => handleClick}>
        Click Me!
      </button>
    </div>
  );
};

describe("ErrorBoundary", () => {
  const wrapper = mount(
    <ErrorBoundary fallbackUI={<ErrorMssg />}>
      <ErrorThrower />
    </ErrorBoundary>
  );
  it("before error is thrown", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("after the error is thrown", () => {
    wrapper.find("button.test-button").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });
});
