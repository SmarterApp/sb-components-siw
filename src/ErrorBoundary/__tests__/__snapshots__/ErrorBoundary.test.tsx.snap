// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary after the error is thrown 1`] = `
<ErrorBoundary
  fallbackUI={<ErrorMssg />}
>
  <ErrorThrower>
    <div>
      <button
        className="test-button"
        onClick={[Function]}
      >
        Click Me!
      </button>
    </div>
  </ErrorThrower>
</ErrorBoundary>
`;

exports[`ErrorBoundary before error is thrown 1`] = `
<ErrorBoundary
  fallbackUI={<ErrorMssg />}
>
  <ErrorThrower>
    <div>
      <button
        className="test-button"
        onClick={[Function]}
      >
        Click Me!
      </button>
    </div>
  </ErrorThrower>
</ErrorBoundary>
`;
